### DISCUSSION TITLE
Setting Serial1 or Serial2 disables Serial

### DISCUSSION THREAD

**Discussion by Hobietime** [earlephilhower/arduino-pico#1306] [Q&A] [answered]
Hello.

I ran into an issue with the Seeed xiao rp2040.

When set up Serial1 or Serial 2 with pins, my USB Serial port disappears. I am only able to reprogram after holding the boot button.

```
// this works
  void setup()
  {
	  
	  //Serial2.setRX(D10);
	  //Serial2.setTX(D8);
	  Serial2.begin(9600);
	  Serial.begin(115200);
  }
  
  void loop()
  {	
  
	  uint8_t buf[8] = {0x01,0x03,0x00,0x00,0x00,0x02,0xc4,0x0b};
	  Serial2.write(buf, 8);
          delay(100);
  }

```
```
// this does not
  void setup()
  {
	  
	  Serial2.setRX(D10);
	  Serial2.setTX(D8);
	  Serial2.begin(9600);
	  Serial.begin(115200);
  }
  
  void loop()
  {	
  
	  uint8_t buf[8] = {0x01,0x03,0x00,0x00,0x00,0x02,0xc4,0x0b};
	  Serial2.write(buf, 8);
          delay(100);
  }
```

I also checked D8 to see if there was any output on  a scope and there is none. Is there something I am doing wrong when initializing Serial1?

I get no compile errors or warnings.

**Comment by earlephilhower** [comment, marked as answer]
Those are illegal pins.  The pico is crashing and it's happening even before the USB port comes up so you don't get the panic message.  Only certain HW pins are allowed for specific peripherals.  See the datasheet or this image:

![image](https://user-images.githubusercontent.com/11875/225388428-e5652f12-b470-4461-930c-e9094df3d86c.png)

**Reply by Hobietime** [reply]
Ah I see!

Thank you.
